dspace-api/src/main/java/org/dspace/discovery/SolrServiceCommunityCollectionItemFilterPlugin.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.discovery;
+
+import java.sql.SQLException;
+
+import org.apache.solr.common.SolrInputDocument;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.Item;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.core.Context;
+import org.dspace.discovery.indexobject.IndexableDSpaceObject;
+
+public class SolrServiceCommunityCollectionItemFilterPlugin implements SolrServiceIndexPlugin {
```

Comment:
Please add JavaDocs to describe the purpose of this plugin.  It likely should say something like "Indexes the UUID of the parent object for any Community, Collection and Item."

I'd also recommend renaming this plugin to be `SolrServiceParentObjectIndexingPlugin`, as the would better describe its purpose.  It's not really filtering Communities, Collections or Items.  Instead it's indexing the parent object for these DSOs.

Reply:
done here https://github.com/DSpace/DSpace/pull/2707/commits/9ce5c586269ebdde32e87d1d7591be15734f7dd3#diff-9f487ddbf8c05ea7008d57b478507f1d